Testnet/MintableTokenInvoice/MintableTokenInvoice/MintableTokenInvoice.cs

```diff
@@ -90,27 +90,21 @@ public string GetInvoiceReference(string transactionReference)
         return $"INV-{invoiceReference.Substring(0, 4)}-{invoiceReference.Substring(4, 4)}-{invoiceReference.Substring(8, 4)}";
     }
 
-    private string ValidateKYC(Address sender)
+    private void EnsureKYCdUserOnly(Address sender)
     {
         // KYC check. Call Identity contract.
         ITransferResult result = this.Call(IdentityContract, 0, "GetClaim", new object[] { sender, KYCProvider });
-        if (!(result?.Success ?? false))
-        {
-            return "Could not determine KYC status";
-        }
+        Assert(result?.Success ?? false, "Could not determine KYC status");
 
         // The return value is a json string encoding of a Model.Claim object, represented as a byte array using ascii encoding.
         // The "Key" and "Description" fields of the json-encoded "Claim" object are expected to contain "Identity Approved".
-        if (result.ReturnValue == null || !Serializer.ToString((byte[])result.ReturnValue).Contains("Identity Approved"))
-        {
-            return "Your KYC status is not valid";
-        }
-
-        return string.Empty;
+        Assert(result.ReturnValue != null && Serializer.ToString((byte[])result.ReturnValue).Contains("Identity Approved"), "Your KYC status is not valid");
```

Comment:
Yeah this seems true because this is how handled in ICOContract as below

```
var result = Call(KYCAddress, 0, "GetClaim", new object[] { identityAddress, (uint)3 /*shufti kyc*/ });
Assert(result.Success && result.ReturnValue != null, "Your KYC is not verified.");
```

Reply:
Done

Reply:
On closer inspection the `result.ReturnValue` can never be null due to:

```
        public byte[] GetBytes(byte[] key)
        {
            byte[] bytes = this.persistenceStrategy.FetchBytes(this.ContractAddress, key);

            if (bytes == null)
                return new byte[0];

            return bytes;
        }        
```